chainer/function_node.py

```diff
@@ -837,19 +838,13 @@ def grad(outputs, inputs, grad_outputs=None, grad_inputs=None, set_grad=False,
     return ret
 
 
-def _backprop(outputs, inputs, grad_required, retain_grad, grads, loss_scale):
-    candidate_funcs, push_candidate, pop_candidate = _get_ordered_func_heap()
-
-    for y in outputs:
-        creator = y.creator_node
-        if creator is not None:
-            push_candidate(creator)
+def _backprop(outputs, inputs, visited_funcs, visited_vars,
+              retain_grad, grads, loss_scale):
```

Comment:
The names `visited_funcs` and `visited_vars` might be misleading in the scope of this function.
How about other naming? Some candidates I'v come up with:

* `subgraph_funcs`, `subgraph_vars`
* `graph_funcs`, `graph_vars`